Services/Administration/OJS.Services.Administration.Business/Contests/ContestsBusinessService.cs

```diff
@@ -266,75 +271,88 @@ public override async Task<ContestAdministrationModel> Create(ContestAdministrat
         return model;
     }
 
-    public async Task TransferParticipantsToPracticeById(int contestId)
-    {
-        var contest = await this.contestsData.GetByIdWithParticipantsScoresAndSubmissions(contestId);
+    public async Task TransferParticipantsToPracticeById(int contestId) =>
+        await this.transactions.ExecuteInTransaction(async () =>
```

Comment:
Make sure only the code that has to be executed in transaction is put inside it (getting the contest for example does not need to be in the transaction scope), otherwise we might prolong it unnecessary. Transactions should be as small and lightweight as possible, because the longer a transaction is open, the longer resources are locked, which can impact the performance of the database.
Also I think you might not need the `ExecuteInTransaction` method at all if instead of the custom `.Delete` methods you call `DeleteMany` directly and then `SaveChanges` after both deletions.

Reply:
Done.